test/ERC4626.t.sol

```diff
@@ -80,3 +80,97 @@ contract SparkVaultERC4626Test is ERC4626Test, SparkVaultTestBase {
     }
 
 }
+
+contract SparkVaultERC4626Test is SparkVaultTestBase {
+
+    address user1 = makeAddr("user1");
+
+    // Do a deposit to get non-zero state
+    function setUp() public override {
+        super.setUp();
+
+        vm.prank(admin);
+        vault.setSsrBounds(ONE_PCT_SSR, FOUR_PCT_SSR);
+
+        vm.prank(setter);
+        vault.setSsr(FOUR_PCT_SSR);
+
+        deal(address(asset), user1, 1_000_000e6);
+
+        vm.startPrank(user1);
+        asset.approve(address(vault), 1_000_000e6);
+        vault.deposit(1_000_000e6, user1);
+        vm.stopPrank();
+
+        skip(1 days);
+    }
+
+    function test_previewRedeem_revertsOverLiquidityBoundary() public {
+        // Deal value accrued to the vault
+        deal(address(asset), address(this), 107.459782e6);
+        asset.transfer(address(vault), 107.459782e6);
+
+        uint256 shares = vault.balanceOf(user1);
+
+        assertEq(shares, 1_000_000e6);
+
+        // Take 1 wei of liquidity from the vault
+        vm.prank(taker);
+        vault.take(1);
+
+        // Redeem should revert over the liquidity boundary
+        vm.expectRevert("SparkVault/insufficient-liquidity");
+        vault.previewRedeem(shares);
+
+        // Transfer 1 wei of liquidity back to the vault
+        vm.prank(taker);
+        asset.transfer(address(vault), 1);
+
+        // Redeem should now succeed
+        vault.previewRedeem(shares);
+    }
+
+    function test_previewWithdraw_revertsOverLiquidityBoundary() public {
+        // Deal value accrued to the vault
+        deal(address(asset), address(this), 107.459782e6);
+        asset.transfer(address(vault), 107.459782e6);
+
+        uint256 assets = vault.assetsOf(user1);
+
+        assertEq(assets, 1_000_107.459782e6);
+
+        // Take 1 wei of liquidity from the vault
+        vm.prank(taker);
+        vault.take(1);
+
+        vm.expectRevert("SparkVault/insufficient-liquidity");
+        vault.previewWithdraw(assets);
+
+        // Transfer 1 wei of liquidity back to the vault
+        vm.prank(taker);
+        asset.transfer(address(vault), 1);
+
+        vault.previewWithdraw(assets);
+    }
+
+    function test_maxRedeem_liquidityLessThanAmount() public {
+        // Remove all liquidity from the vault
+        vm.startPrank(taker);
+        vault.take(asset.balanceOf(address(vault)));
+        vm.stopPrank();
+
+        // Redeem max amount should return 0
+        assertEq(vault.maxRedeem(user1), 0);
```

Comment:
Add case where there is a some liquidity to demonstrate that the user can withdraw some